node/lifecycle_test.go

```diff
@@ -309,7 +289,7 @@ func wireUpSystem(ctx context.Context, provider PodLifecycleHandler, f testFunct
 
 	// Shutdown the pod controller, and wait for it to exit
 	cancel()
-	return <-sys.retChan
+	return nil
```

Comment:
Aren't we missing out on checking that the pod controller had no errors at shutdown? (and shutdown successfully?)

Reply:
Right now there's nothing to wait on.
 We can certainly do that... e.g. taking from context.Context we could impement `<-pc.Done()` and `pc.Err()`.
However if we do this we should wait on workers and everything to stop before returning.
WDYT?

Reply:
I think it makes sense to. I had a PR that put the workers in an errgroup. Perhaps you can edd the API for now, and then we can actually plumb it in later.